cmd/controller-manager/app/controller_manager.go

```diff
@@ -90,19 +90,20 @@ func Run(controllerManagerOptions *options.ControllerManagerServer) error {
 	// Build the K8s kubeconfig / client / clientBuilder
 	glog.V(4).Info("Building k8s kubeconfig")
 
-	k8sKubeconfig, err := rest.InClusterConfig()
+	var err error
+	var k8sKubeconfig *rest.Config
+	if controllerManagerOptions.K8sAPIServerURL == "" && controllerManagerOptions.K8sKubeconfigPath == "" {
+		k8sKubeconfig, err = rest.InClusterConfig()
+	} else {
+		k8sKubeconfig, err = clientcmd.BuildConfigFromFlags(
+			controllerManagerOptions.K8sAPIServerURL,
+			controllerManagerOptions.K8sKubeconfigPath)
```

Comment:
I totally agree that it's error prone. That's why I want there to be one place with all the errors. 

I am totally lazy. I'd rather someone whos already done the work (`kubectl`) to provide me a way to base off of that work if I want to use a client with a defaulted kubeconfig that doesn't require me to know anything about kube other than, "there is a default place to store configs, and this function XYZ()" exposes it. Does not sound like there is such a thing, unfortunately.